src/12_integrate_HIV.py

```diff
@@ -21,43 +21,41 @@
     .config("spark.network.timeout","10000000ms") \
     .getOrCreate()
 
-stg = os.makedirs("staging/HIV", exist_ok=True)
+os.makedirs("staging/HIV", exist_ok=True)
 data = biobricks.assets("MoleculeNet")
 
-# BUILD COMPOUNDS =========================================================
+# Build compounds
 cmpraw = spark.read.parquet(data.HIV_parquet)
-
-# Filter out rows with null SMILES values
-cmp1 = cmpraw.filter(F.col('smiles').isNotNull())
-cmp2 = cmp1.withColumn('sid', F.monotonically_increasing_id().cast('string'))
-
-# Create the substances table in JSON format
-subjson = cmp2.select("sid", F.to_json(F.struct("smiles", "activity", "HIV_active")).alias("data"))
+cmp_filtered = cmpraw.filter(F.col('smiles').isNotNull())
+cmp_with_sid = cmp_filtered.withColumn('sid', F.monotonically_increasing_id().cast('string'))
+
+# Convert SMILES to InChI
+smiles_to_inchi_udf = get_smiles_to_inchi_udf()
+cmp_with_inchi = cmp_with_sid.withColumn("inchi", smiles_to_inchi_udf(F.col("smiles"))).cache()
+
+# Substances table (only structural information)
+subjson = cmp_with_inchi.select(
+    "sid", 
+    F.to_json(F.struct("smiles", "inchi")).alias("data")
+)
 subjson.write.mode("overwrite").parquet("staging/HIV/substances.parquet")
 
-# WRITE PROPERTIES =====================================================
-# Define the JSON data for properties
+# Properties table
 data_json = json.dumps({
-    'HIV_active = 1': 'positive',
-    'HIV_active = 0': 'negative'
+    'property': 'HIV_active',
+    'active_value': 1,
+    'inactive_value': 0
 })
-
-# Create the DataFrame with a single row
-properties = spark.createDataFrame([
-    (0, data_json)
-], ["pid", "data"])
-
-# Write the properties table to a Parquet file
+properties = spark.createDataFrame([("0", data_json)], ["pid", "data"])
 properties.write.mode("overwrite").parquet("staging/HIV/properties.parquet")
 
-# BUILD ACTIVITIES ======================================================
-# Set the source and PID for the activity table
-cmp_with_ids = cmp2.withColumn("aid", F.col("sid")) \
-                   .withColumn("pid", F.lit(0)) \
-                   .withColumn("source", F.lit("HIV"))
-
-# Final Activity Table: select required columns including smiles
-activity_table = cmp_with_ids.select("aid", "pid", "sid", "smiles", "source", "HIV_active")
+# Activities table
+cmp_with_ids = cmp_with_inchi.withColumn("aid", F.col("sid")) \
+                             .withColumn("pid", F.lit("0")) \
+                             .withColumn("source", F.lit("HIV")) \
+                             .withColumnRenamed("HIV_active", "value")
+activity_table = cmp_with_ids.select("aid", "pid", "sid", "smiles", "inchi", "source", "value")
+activity_table.write.mode("overwrite").parquet("staging/HIV/activities.parquet")
 
-# Write the activities table to a Parquet file
-activity_table.write.mode("overwrite").parquet("staging/HIV/activities.parquet")
\ No newline at end of file
+# Clean up
+cmp_with_inchi.unpersist()
```

Comment:
This seems unnecessary. If you want to clean up the spark environment probably just kill spark. `spark.stop()`